### PR TITLE
Refactor/minor

### PR DESCRIPTION
**Pull request by ZumZoom**

### CODE REVIEW

File: contracts/extensions/ExtensionLib.sol [hunk 1]
```diff
@@ -0,0 +1,21 @@
+// SPDX-License-Identifier: MIT
+
+pragma solidity ^0.8.0;
+
+library ExtensionLib {
```
Comment (byshape) [inline contracts/extensions/ExtensionLib.sol:5]:
```suggestion
/**
 * @title Extension Library
 * @notice Library to retrieve data from the bitmap.
 */
library ExtensionLib {
```

File: contracts/extensions/ExtensionLib.sol [hunk 2]
```diff
@@ -0,0 +1,21 @@
...
+    bytes1 private constant _RESOLVER_FEE_FLAG = 0x01;
+    bytes1 private constant _INTEGRATOR_FEE_FLAG = 0x02;
+    uint256 private constant _WHITELIST_SHIFT = 3;
+
+    function resolverFeeEnabled(bytes calldata extraData) internal pure returns (bool) {
```
Comment (byshape) [inline contracts/extensions/ExtensionLib.sol:10]:
```suggestion
    /**
     * @notice Checks if the resolver fee is enabled
     * @param extraData Data to be processed in the extension
     * @return True if the resolver fee is enabled
     */
    function resolverFeeEnabled(bytes calldata extraData) internal pure returns (bool) {
```

File: contracts/extensions/ExtensionLib.sol [hunk 3]
```diff
@@ -0,0 +1,21 @@
...
+        return extraData[extraData.length - 1] & _RESOLVER_FEE_FLAG == _RESOLVER_FEE_FLAG;
+    }
+
+    function integratorFeeEnabled(bytes calldata extraData) internal pure returns (bool) {
```
Comment (byshape) [inline contracts/extensions/ExtensionLib.sol:14]:
```suggestion
    /**
     * @notice Checks if the integrator fee is enabled
     * @param extraData Data to be processed in the extension
     * @return True if the integrator fee is enabled
     */
    function integratorFeeEnabled(bytes calldata extraData) internal pure returns (bool) {
```

File: contracts/extensions/ExtensionLib.sol [hunk 4]
```diff
@@ -0,0 +1,21 @@
...
+        return extraData[extraData.length - 1] & _INTEGRATOR_FEE_FLAG == _INTEGRATOR_FEE_FLAG;
+    }
+
+    function whitelistCount(bytes calldata extraData) internal pure returns (uint256) {
```
Comment (byshape) [inline contracts/extensions/ExtensionLib.sol:18]:
```suggestion
    /**
     * @notice Gets the whitelist length
     * @param extraData Data to be processed in the extension
     * @return The whitelist length
     */
    function whitelistLength(bytes calldata extraData) internal pure returns (uint256) {
```

File: contracts/extensions/WhitelistExtension.sol [hunk 5]
```diff
@@ -62,10 +65,11 @@ abstract contract WhitelistExtension is BaseExtension {
         uint256 remainingMakingAmount,
         bytes calldata extraData
     ) internal virtual override {
-        uint8 resolversLength = uint8(extraData[extraData.length - 1]) >> 3;
-        uint256 whitelistLength = 4 + resolversLength * 12;
-        bytes calldata whitelist = extraData[:whitelistLength];
-        if (!_isWhitelisted(whitelist, resolversLength, taker)) revert ResolverIsNotWhitelisted();
-        super._postInteraction(order, extension, orderHash, taker, makingAmount, takingAmount, remainingMakingAmount, extraData[whitelistLength:]);
+        uint256 resolversCount = extraData.whitelistCount();
```
Comment (byshape) [inline contracts/extensions/WhitelistExtension.sol:68]: It feels like `whitelistLength` is a more suitable name for the function, wdyt?

**Review by byshape — APPROVED** [review]

File: contracts/extensions/WhitelistExtension.sol (hunk 5 above)
Comment (ZumZoom) [inline contracts/extensions/WhitelistExtension.sol:68]:
```suggestion
        uint256 resolversCount = extraData.resolversCount();
```

**Pull request merged**